### ISSUE TITLE
<About embedding different number of sampling points in the loss function>

### ISSUE THREAD

**Issue by zyz-rise** [lululxvi/deepxde#440]
Dear Dr. Lu，
         Thanks for the provision of DeepXDE library , which greatly facilitates the use of PINN to solve differential equations！
         I wonder if you could help me solve the following problems：

**1. Calculation of loss function**

> Consider the following system of ODEs:

```
 def ode_system(t, x):
       epsilon=0.05
       x1, x2, x3 = x[:, 0:1], x[:, 1:2], x[:, 2:3]
       dx1_t = dde.grad.jacobian(x, t, i=0)
       dx2_t = dde.grad.jacobian(x, t, i=1)   
       dx3_t = dde.grad.jacobian(x, t, i=2)
  return [dx1_t - x2/epsilon - x1 - 2 * x3, dx2_t + x1/epsilon - x2, dx3_t - x1**2 - x2**2]
```
I set the number of sampling points to 200, that is, the neural network generates 200 output data points for x1, x2, and x3, which are embedded in the loss function to calculate the MSE. **I wonder if PINN can use 200 points for x1, 100 points for x2, and 50 points for x3 when calculating the MSE of loss function?** But it seems that num_domain can only set one value, instead of a list. So can you tell me what should I do? Need to modify the source code?
```
geom = dde.geometry.TimeDomain(0.0, 1.0)
ic1 = dde.IC(geom, lambda x: 1, boundary, component=0)
ic2 = dde.IC(geom, lambda x: 0, boundary, component=1)
ic3 = dde.IC(geom, lambda x: 0, boundary, component=2)
data = dde.data.PDE(geom, ode_system, [ic1, ic2, ic3], num_domain=200, 2, num_test=100)
```
**2. About the generalization ability of the model**
         The model only performed well in the training domain (0.0 , 1.0). It seems that there was overfitting. I wonder if it is due to the hyperparameter settings such as the number of layers and the number of neurons in the network?
```
layer_size = [1] + [64] * 6 + [3]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.maps.FNN(layer_size, activation, initializer)
model = dde.Model(data, net)
model.compile("adam", lr=0.001)
start_time = time.time() 
loss_history, train_state = model.train(epochs=50000)
```

![Figure 2021-12-02 205212](https://user-images.githubusercontent.com/95417257/144427004-c6fd31f6-6884-49be-8371-7e2c799c7320.png)

**Comment by lululxvi** [comment]
No. If you need to use different training points for different PDE terms, you have to modify the code. Why do you need this?

If you want to do extrapolation, then you need to have the training points for that region.

**Comment by zyz-rise** [comment]
im glad to receive your reply :-)
     In terms of the solution form of ODEs, the third term x3 is a slow variable. So I think x3 does not need too many training points, but x1 and x2 may be needed... just want to verify it. And do you have any advices for modifying the code or anything else? Thanks and best wishes~

**Comment by lululxvi** [comment]
It is not a problem to the speed. If you want to modify the code, you need to modify the class `PDE`.
